src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -132,52 +133,58 @@ public void testAbbreviationsAreRejected() {
     public void testOptionalOnlyUsage() {
         final OptionalOnlyArguments oo = new OptionalOnlyArguments();
         final CommandLineArgumentParser clp = new CommandLineArgumentParser(oo);
-        final String out = clp.usage(false); // without common args
+        final String out = clp.usage(false, false); // without common/hidden args
         final int reqIndex = out.indexOf("Required Arguments:");
         Assert.assertTrue(reqIndex < 0);
         Assert.assertTrue(out.indexOf("Optional Arguments:", reqIndex) > 0);
         Assert.assertEquals(out.indexOf("Conditional Arguments:", reqIndex), -1);
+        Assert.assertEquals(out.indexOf("Advanced Arguments:", reqIndex), -1);
     }
 
     /**
      * Validate the text emitted by a call to usage by ensuring that required arguments are
      * emitted before optional ones.
      */
-    private void validateRequiredOptionalUsage(final CommandLineArgumentParser clp, final boolean withDefault) {
-        final String out = clp.usage(withDefault); // with common args
+    private void validateRequiredOptionalUsage(final CommandLineArgumentParser clp, final boolean withDefault, final boolean hasAdvanced) {
+        final String out = clp.usage(withDefault, false); // with common args, without hidden args
         // Required arguments should appear before optional ones
         final int reqIndex = out.indexOf("Required Arguments:");
         Assert.assertTrue(reqIndex > 0);
         Assert.assertTrue(out.indexOf("Optional Arguments:", reqIndex) > 0);
         Assert.assertEquals(out.indexOf("Conditional Arguments:", reqIndex), -1);
+        Assert.assertEquals(out.indexOf("Advanced Arguments:", reqIndex) != -1, hasAdvanced);
     }
 
     @Test
     public void testRequiredOptionalWithDefaultUsage() {
         final FrobnicateArguments fo = new FrobnicateArguments();
         final CommandLineArgumentParser clp = new CommandLineArgumentParser(fo);
-        validateRequiredOptionalUsage(clp, true); // with common args
+        // have the special showHidden advanced argument
+        validateRequiredOptionalUsage(clp, true, true); // with common args
```

Comment:
This is validating if the "Advanced Arguments:" prefix is printed, @cmnbroad. Because one of the special arguments is advanced (`--showHidden`), the `FrobnicateArguments` should print this out.